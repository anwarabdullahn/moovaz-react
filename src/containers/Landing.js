import React, { Component } from 'react';
import Topic from '../components/Topic';
import { Button, Modal, ModalHeader, ModalBody, ModalFooter, Input } from 'reactstrap';
import uuidv4 from 'uuid/v4';

class Landing extends Component {
	constructor() {
		super();
		this.state = {
			topics: [
				{ id: 1111, title: 'First Topic', upvote: 0, downvote: 0 },
				{ id: 2222, title: 'Second Topic', upvote: 0, downvote: 0 }
			],
			modal: false,
			title: ''
		};
	}

	toggle = () => this.setState({ modal: !this.state.modal });

	changeHandle = (e) => this.setState({ [e.target.name]: e.target.value });

	submitHandle = () => {
		const { title, topics } = this.state,
			data = { id: uuidv4(), title, upvote: 0, downvote: 0 };
		this.setState({ topics: [ ...topics, data ], title: '' });
		console.log('data', data);
		this.toggle();
	};

	render() {
		return (
			<div className="mt-5 mx-3">
				<Topic topics={this.state.topics} addHandle={this.toggle} />
				<Modal isOpen={this.state.modal} toggle={this.toggle} className={this.props.className}>
					<ModalHeader toggle={this.toggle}>Topic title</ModalHeader>
					<ModalBody>
						<Input placeholder="title" value={this.state.title} name="title" onChange={this.changeHandle} />
					</ModalBody>
					<ModalFooter>
						<Button color="primary" onClick={this.submitHandle}>
							Add Topic
						</Button>
						<Button color="secondary" onClick={this.toggle}>
							Cancel
						</Button>
					</ModalFooter>
				</Modal>
			</div>
		);
	}
}

export default Landing;
