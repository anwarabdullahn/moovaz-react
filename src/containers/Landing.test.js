import React from 'react';
import { shallow } from 'enzyme';
import Landing from './Landing';
describe('Landing', () => {
	it('should render Landing correctly in "debug" mode', () => {
		expect(shallow(<Landing debug />)).toMatchSnapshot();
	});
});
