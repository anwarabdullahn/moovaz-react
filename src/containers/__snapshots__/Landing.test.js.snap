// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing should render Landing correctly in "debug" mode 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`Landing should render correctly in "debug" mode 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
